From 5d2c21f392240dca0e5428f6cc56425723f134a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 30 Jul 2025 15:38:46 +0200
Subject: [PATCH] Fix handling file not found error

If the file does not exists, don't call `stat()` as it causes a `FileNotFoundError` exception.
---
 src/window.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/window.py b/src/window.py
index be09423..9977c8f 100644
--- a/src/window.py
+++ b/src/window.py
@@ -413,9 +413,12 @@ def compress_filenames(self, filenames):
                 error_message = _("This file doesn't exist.").format(filename)
 
             # Check format
-            size = path.stat().st_size
-            if not self.check_format(filename) or size <= 0:
-                error_message = _("Format of this file is not supported.").format(path.name)
+            if not error_message:
+                size = path.stat().st_size
+                if not self.check_format(filename) or size <= 0:
+                    error_message = _("Format of this file is not supported.").format(path.name)
+            else:
+                size = 0
 
             if not error_message:
                 new_filename = self.create_new_filename(path)
